# symbiont-sra
Data from [Gerth & Hurst 2017](http://dx.doi.org/10.7717/peerj.3529)

This repository was created to enable interested readers to reproduce the steps described in [Gerth & Hurst 2017](https://github.com/gerthmicha/symbiont-sra/). A guide for the steps used to retrieve *Apis* SRA data, and to extract microbial information is given in [guide.md](https://github.com/gerthmicha/symbiont-sra/blob/master/guide.md). All files needed to follow this guide (including any intermediate files) can be found in the folder '[example](https://github.com/gerthmicha/symbiont-sra/tree/master/example)', and are described in [guide.md](https://github.com/gerthmicha/symbiont-sra/blob/master/guide.md). 

Furthermore, all supermatrix files for phylogenomic analyses described in our study are accessible in the folder '[alignments](https://github.com/gerthmicha/symbiont-sra/tree/master/alignments)'. Two alignment files created with [SSU-ALIGN](http://eddylab.org/software/ssu-align/) for *Lactobacillus* 16S phylogenetic analysis are also found in this directory. 
The folder '[genomes](https://github.com/gerthmicha/symbiont-sra/tree/master/genomes)' contains annotated genomes of *Apis* symbionts that we retrieved and assembled from 'contaminated' short read data.
All reference sequences that were used in our screen may be found under '[references](https://github.com/gerthmicha/symbiont-sra/tree/master/references)'.
